vm-presets.md

```diff
@@ -0,0 +1,237 @@
+Virtual Machine Presets
+=============================
+
+`VirtualMachinePresets` are an extension to general `VirtualMachine`
+Configuration based heavily on `PodPresets` from Kubernetes. When a
+`VirtualMachine` is created, any applicable `VirtualMachinePresets`
+will be applied to the existing spec for the `VirtualMachine`. This allows
+for re-use of common settings that should apply to multiple `VirtualMachines`.
+
+Usage
+------------------------
+
+KubeVirt uses Kubernetes `Labels` to determine which `VirtualMachinePresets`
+apply to any given `VirtualMachine`, similarly to how `PodPresets` work in
+Kubernetes. The `VirtualMachine` is marked with an Annotation upon successful
+completion.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: VirtualMachine
+metadata:
+  name: testvm
+  selector:
+    matchLabels:
+      flavor: large-vm
+spec:
+  ...
+
+```
+
+Any domain structure can be listed in the `spec` of a `VirtualMachinePreset`.
+e.g. Clock, Features, Memory, CPU, or Devices such network interfaces or disks.
+All elements of the `spec` section of a `VirtualMachinePreset` will be applied
+to the `VirtualMachine`. Single element structures will be applied individually
+while members of lists will be appended. See the `Examples` section for more
+detail.
+
+
+Once a `VirtualMachinePreset` is successfully applied to a `VirtualMachine`,
+the `VirtualMachine` will be marked with an annotation to indicate that it
+was applied. If an error occurs while a `VirtualMachinePreset` is being applied
+(for example, if a conflict occurs), none of the `VirtualMachinePreset` will be
+applied.
```

Comment:
How and where are errors shown? I.e. in case thats oemthing can not be applied?

Reply:
+1 on adding that. I would expect the same like in https://kubernetes.io/docs/tasks/inject-data-application/podpreset/#conflict-example. So when I do a `describe`, I will see an event for every merge conflict.

Reply:
Sounds good, we need to make sure to cover this in teh api design doc as well.

Here we can be more reöaxed and provide an example and reference PodPresets as Roman suggests.